package NYT.NHydra.NProto;

import "yt_proto/yt/core/misc/proto/guid.proto";

////////////////////////////////////////////////////////////////////////////////

message TSnapshotMeta
{
    reserved 1, 9;
    optional int64 sequence_number = 2;
    optional uint64 random_seed = 3;
    optional uint64 state_hash = 4;
    optional uint64 timestamp = 5;
    optional int32 last_segment_id = 6;
    optional int32 last_record_id = 7;
    optional int32 last_mutation_term = 8;
    optional int32 last_mutation_reign = 11;
    optional bool read_only = 10;
    optional uint64 logical_time = 12;
}

message TChangelogMeta
{ }

////////////////////////////////////////////////////////////////////////////////

message TMutationHeader
{
    required string mutation_type = 1;
    required uint64 timestamp = 2;
    required uint64 random_seed = 3;
    required int32 segment_id = 5;
    required int32 record_id = 6;
    optional NYT.NProto.TGuid mutation_id = 7;
    optional int32 reign = 8 [default = 0];
    required uint64 prev_random_seed = 9;
    required int64 sequence_number = 10;
    optional int32 term = 11 [default = 0];
}

////////////////////////////////////////////////////////////////////////////////

message TVoidMutationResponse
{ }

////////////////////////////////////////////////////////////////////////////////
